app/Actions/Fortify/CreateNewUser.php

```diff
@@ -60,7 +61,9 @@ public function create(array $input): User
                 }
 
                 if (array_key_exists('company_name', $input)) {
-                    $this->createTeam(
+                    $attributes = $this->prepareAttributes($input);
```

Comment:
Why not move attributes preparation to the `createCompany` method body. Then, you only need to call the method here like: `$this->createCompany($user, $input);`. The method itself is then responsible for preparing attributes and choosing which of the input is needed to create a company properly